lib/api/core/plugins/pluginContext.js

```diff
@@ -150,66 +150,92 @@ class PluginContext {
 }
 
 /**
- * @this {Kuzzle}
+ * @param {Kuzzle} kuzzle 
  * @param {Request} request
- * @param {function} callback
+ * @param {boolean} [overloadProtect]
+ * @param {Function} [callback]
  */
-function execute (request, callback) {
-  const kuzzle = this;
-
-  return kuzzle.funnel.processRequest(request)
-    .then(response => {
-      if (callback) {
-        try {
-          callback(null, response);
-        }
-        catch (e) {
-          kuzzle.pluginsManager.trigger('log:error', new PluginImplementationError(`Uncatched error by a plugin callback: ${e}`));
-        }
+function execute (kuzzle, request, overloadProtect = true, callback = null) {
+  let resolve = null,
+    reject = null, 
+    deferred = null;
 
-        return;
-      }
+  let error = null;
+  const overloadProtectType = typeof overloadProtect;
+
+  if (overloadProtectType === 'function') {
+    callback = overloadProtect;
+    overloadProtect = true;
```

Comment:
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png 'Severity: MINOR') Introduce a new variable instead of reusing the parameter "overloadProtect". [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://sonarqube.kaliop.net/coding_rules#rule_key=javascript%3AS1226)